### PR TITLE
Logger rate limiting

### PR DESCRIPTION
**Pull request by nsavoire**
# What does this PR do?

Implement log rate limiting when log level is above debug. Rate limit is set at 100 log/s.

### CODE REVIEW

File: include/ratelimiter.hpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  IntervalRateLimiter(uint64_t max_count_per_interval,
+                      std::chrono::nanoseconds interval) noexcept
+      : _max_count_per_interval(max_count_per_interval), _interval(interval) {}
+
+  bool check() {
+    auto old_count = _count.fetch_add(1, std::memory_order_acq_rel);
+    if (old_count < _max_count_per_interval) {
+      return true;
+    }
+    return check_slow();
```
Comment (r1viollet) [inline include/ratelimiter.hpp:45]: should we not change slightly this pattern to avoid checking at every log ?

**Review by r1viollet — APPROVED** [review]
LGTM
A good addition
I think we could still make it more efficient in heavy logging cases, though this is probably only useful in pathological cases that never occur.

File: include/logger_setup.hpp [hunk 2]
```diff
@@ -5,6 +5,13 @@
 
 #pragma once
 
+#include <cstdint>
+
 namespace ddprof {
-void setup_logger(const char *log_mode, const char *log_level);
-}
+inline constexpr auto kMaxLogPerSecForNonDebug = 100;
```
Comment (r1viollet) [inline include/logger_setup.hpp:11]: ok for 100 :+1:

**Review by r1viollet — APPROVED** [review]
LGTM

File: include/ratelimiter.hpp (hunk 1 above)
Comment (r1viollet) [reply include/ratelimiter.hpp:45]: Never mind, this is a pathological case, we should just avoid logging so much.

**Review by r1viollet — APPROVED** [review]
LGTM

**Review by sanchda — DISMISSED** [review]
LGTM--thank you for the improvement!

**Review by sanchda — APPROVED** [review]

**Pull request merged**